piptools/utils.py

```diff
@@ -695,11 +686,27 @@ def parse_config_file(config_file: Path) -> dict[str, Any]:
             hint=f"Could not parse '{config_file !s}': {value_err !s}",
         )
 
+    cli_opts = {
+        opt: option
+        for option in click_context.command.params
+        for opt in itertools.chain(option.opts, option.secondary_opts)
```

Comment:
```suggestion
        for opt in itertools.chain(option.opts, option.secondary_opts)
        if opt.statrswith("--")
```

Better to filter out short option names, because we use long option names in config.

```
 '-P': <Option upgrade_packages>,
 '-U': <Option upgrade>,
 '-f': <Option find_links>,
 '-i': <Option index_url>,
 '-n': <Option dry_run>,
 '-o': <Option output_file>,
 '-p': <Option pre>,
 '-q': <Option quiet>,
 '-r': <Option rebuild>,
 '-v': <Option verbose>,
 ```